flask_resty/spec/spec_declaration.py

```diff
@@ -0,0 +1,9 @@
+class SpecDeclaration(object):
+
+    def update_spec(self, path, spec, **kwargs):
+        """When overridden, gives a chance to update the path fragment of the
+        specification
+        :param dict path: swagger-spec/blob/master/versions/2.0.md#pathItemObject) # noqa
+        :param apispec.APISpec spec: the specification object
+        """
+        pass
```

Comment:
To elaborate - I think when we call into `update_spec`, we should consider using duck typing instead of doing an actual type check... probably more Pythonic that way.